hazelcast/client.py

```diff
@@ -50,6 +51,13 @@ class HazelcastClient(object):
     distributed data structures on the Hazelcast clusters.
 
     Keyword Args:
+        config (`hazelcast.config.Config`): Optional configuration object
```

Comment:
Could `hazelcast.config.Config` inherit from `dict`?
or have a `to_dict()`/`from_dict()` method.

Then you could use dict unpacking to create the client with the config.
You could probably do this instead of accepting a `Config` object directly or make just another option.

Either way, it would make the transition from v3 easier.

```python
# lets log the config settings
LOGGER.info(my_config_obj.dict())

HazelcastClient(**my_config_obj.dict())
```